target/linux/generic/files/drivers/platform/mikrotik/mikrotik_wlan_lz77.c

```diff
@@ -0,0 +1,443 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+/*
+ * Copyright (C) 2023 John Thomson
+ */
+
+#include <linux/module.h>
+#include <linux/slab.h>
+#include <linux/string.h>
+#include <linux/errno.h>
+
+#include "mikrotik_wlan_lz77.h"
+
+#define MIKRO_LZ77 "[mikrotik wlan lz77] "
```

Comment:
AFAICT the content of that file has become a pure LZ77 decompressor, so you might want to:
- rename the file
- change function prefixes
- change this print prefix (ideally to something a bit shorter as well, so it doesn't take too much terminal real-estate)

Maybe "rb_lz77" to keep in line with the "rb" prefix of other bits of code in this folder and keep the identifier prefixes short enough? (`mikrotik_wlan_lz77_*` is quite a "mouthful" imho :).

That being said, I still wonder if this should be submitted upstream as a standalone lz77 decoder? In which case the prefix would naturally be "lz77_".